scripts/Interface/ui/sans_isis/sans_data_processor_gui.py

```diff
@@ -484,6 +480,7 @@ def _data_changed(self, row_location, column, old_value, new_value):
     def _row_inserted(self, row_location):
         if row_location.depth() > 1:
             self.data_processor_table.removeRowAt(row_location)
+        self._call_settings_listeners(lambda listener: listener.on_insert_row())
```

Comment:
Looks good and fixes the bug - haven't seen this before, normally I would send a signal and that would be connected to a slot in the presenter - here it looks like you keep track of all the listeners and explicitly call a method of all listeners after an event - is there any advantage to this? One problem might be that you might want multiple listeners (but for different things) such that for example not all of them have a method called `on_insert_row()`

Reply:
As discussed offline @DavidFair is already planning to look at this further in maintenance. I'm very happy for this to get merged.